codegen/smithy-aws-ruby-codegen/src/main/java/software/amazon/smithy/aws/ruby/codegen/AWSProtocols.java

```diff
@@ -97,22 +96,22 @@ metadata service (IMDS) on an EC2 instance.
 
         String identityProviderChain = "nil"; // TODO: Add the actual credential/identity provider chain
         String defaultIdentity = "%s.new(access_key_id: 'stubbed-akid', secret_access_key: 'stubbed-secret')"
-                .formatted(SIGV4_IDENTITY);
+                .formatted(AWS_CREDENTIALS_IDENTITY);
         String defaultConfigValue = "cfg[:stub_responses] ? %s.new(proc { %s }) : %s"
-                .formatted(Hearth.IDENTITY_RESOLVER, defaultIdentity, identityProviderChain);
+                .formatted(Hearth.IDENTITY_PROVIDER, defaultIdentity, identityProviderChain);
 
-        ClientConfig identityResolverConfig = ClientConfig.builder()
-                .name("sigv4_identity_resolver")
-                .documentationRbsAndValidationType(Hearth.IDENTITY_RESOLVER.toString())
-                .documentation(identityProviderDocumentation)
+        ClientConfig identityProviderConfig = ClientConfig.builder()
+                .name("credential_provider")
```

Comment:
I think this will have to be sigv4_identity_provider right? (sigv4) Credentials is just the name of the Identity